crates/ruff/src/checkers/imports.rs

```diff
@@ -54,5 +55,45 @@ pub fn check_imports(
             &blocks, python_ast, locator, stylist, settings, autofix,
         ));
     }
-    diagnostics
+    let mut imports: FxHashMap<PathBuf, Vec<Import>> = FxHashMap::default();
+    let mut imports_vec = vec![];
+    for &block in &blocks {
+        block.imports.iter().for_each(|&stmt| match &stmt.node {
+            StmtKind::Import { names } => {
+                // from testing, seems this should only have one entry
+                imports_vec.push(Import {
+                    name: names[0].node.name.clone(),
+                    location: stmt.location,
+                    end_location: stmt.end_location.unwrap(),
```

Comment:
Not specific to this PR: @charliermarsh it seems that it is guaranteed that location is always set. Can we introduce a helper function / extension method that calls the `unwrap` rather than sprinkling the unwraps all over our code base?

Reply:
Yeah for context, the Python AST treats end locations as optional. In RustPython, though, we always provide them (and `Expr::new` _requires_ them). So the _interface_ mirrors CPython, but they _are_ guaranteed in practice.